app/controllers/CheckDetailsController.scala

```diff
@@ -33,10 +34,9 @@ import uk.gov.hmrc.http.HeaderCarrier
 import uk.gov.hmrc.play.bootstrap.frontend.controller.FrontendBaseController
 import util.FMNConstants.{EmptyString, FMNOrigin, IVOrigin, PDVOrigin}
 import util.FMNHelper.comparePostCode
-
 import javax.inject.Inject
-import scala.concurrent.{ExecutionContext, Future}
-import scala.util.{Failure, Success, Try}
+import scala.concurrent.{ExecutionContext, Future, Await}
+import scala.concurrent.duration._
 
 class CheckDetailsController @Inject()(
```

Comment:
I do not like how messy this controller is with PDV checks at multiple places in the code. 

I have a refractor of this and I think we can mob to improve this.